### ISSUE TITLE
Slow training on Ray as Compared to Accelerate multi node

### ISSUE THREAD

**Issue by solidate** [axolotl-ai-cloud/axolotl#2818]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behaviour

Axolotl on Ray perform faster distributed training.

### Current behaviour

We are trying to Fine tune Gemma 3 27B model using axolotl on 16xH100 GPUs.
We tried training it with Multi Node on accelerate with 2 machines each having 8xH100s.

But we wanted a more auto-scalable solution to fine tune the model. So we followed the guide [Ray Train](https://docs.axolotl.ai/docs/ray-integration.html)

We were able to launch the fine tuning job to tune Gemma 3 27B on Ray cluster using the same data and noticed that the model is taking more than 4 hrs  to get train only for 1 epoch, while with accelerate with similar Infrastructure spec it is taking <1hr to train for 10 epochs.

Is there any other config that we need to use in order to perform faster training on the ray cluster?
At this moment, The Ray train guide seems incomplete/lack some configs to fully utilize the benefit of a HPC.

![Image](https://github.com/user-attachments/assets/975772d9-d193-4351-a73a-3865d1acf2c8)

### Steps to reproduce

axolotl train config.yaml --use-ray

### Config yaml

```yaml
Below is the config we used to perform training on Ray Cluster:

base_model: google/gemma-3-27b-it
model_type:  AutoModelForCausalLM 
tokenizer_type: AutoTokenizer
# Automatically upload checkpoint and final model to HF
# hub_model_id: username/custom_model_name

load_in_8bit: True
load_in_4bit: False

# bnb_config_kwargs:
#   llm_int8_has_fp16_weight: false

chat_template: gemma3
rl: dpo
datasets:
  - path: /mnt/cluster_storage/data/train.json 
    type: chat_template.default
    field_messages: "prompt" ##messages
    field_chosen: "chosen"
    field_rejected: "rejected"
    message_property_mappings:
      role: role
      content: content
    roles:
      user: ["user"]
      assistant: ["assistant"]
      system: ["system"]

test_datasets:
  - path: /mnt/cluster_storage/data/test.json 
    type: chat_template.default
    field_messages: "prompt" ##messages
    field_chosen: "chosen"
    field_rejected: "rejected"
    message_property_mappings:
      role: role
      content: content
    roles:
      user: ["user"]
      assistant: ["assistant"]
      system: ["system"]

dataset_prepared_path:
#val_set_size: 0.05
output_dir: /mnt/cluster_storage/gemma3_output
use_tensorboard: true

sequence_len: 1024
sample_packing: false
pad_to_sequence_len: true

adapter: lora
lora_model_dir:
lora_r: 16
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
# lora_target_modules: 'language_model.model.layers.[\d]+.(mlp|cross_attn|self_attn).(up|down|gate|q|k|v|o)_proj'

# wandb configuration if you're using it
# Make sure your `WANDB_API_KEY` environment variable is set (recommended) or you login to wandb with `wandb login`.

wandb_project: # Your wandb project name
wandb_entity: # A wandb Team name if using a Team
wandb_watch:
wandb_name: # Set the name of your wandb run
wandb_run_id: # Set the ID of your wandb run
wandb_log_model: 

gradient_accumulation_steps: 1
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002

bf16: auto
tf32: false
# fp16: true

resume_from_checkpoint:
logging_steps: 1
flash_attention: false
deepspeed: /mnt/cluster_storage/deepspeed_configs/zero2.json

warmup_steps: 10
evals_per_epoch: 2
saves_per_epoch: 1
weight_decay: 0.0
shuffle: True
# overrides_of_model_config:
#   rope_scaling:
#     type: "linear"
#     factor: 0.5
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false

profiler_steps: 5

use_ray: true
ray_num_workers: 8
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

axolotl==0.9.2

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by kouroshHakha** [comment]
Hi @solidate , can you provide a fully reproducible path for both ray and accelerate? 

I notice that in your yaml you specified ray_num_workers: 8 but your screen shot uses 16. So there is some discrepancy there. If you can provide a step by step reproduction we can take a look.
